### PR TITLE
Automated some step with ansible

### PR DESCRIPTION
**Pull request by marius-balteanu**
- Updated Debian 
- Set nfs default shared regardless the OS type 
- Move git install to ansible
- Update docker-compose version
- Commented puma gem (is already available in Redmine 4)
- Automated some steps

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -28,9 +28,7 @@ Vagrant.configure(2) do |config|
   end
 
   # Sync the sources folder with the machine.
-  # For Windows `nfs` is preffered due to poor performance of default settings.
-  synced_folder_type = Vagrant::Util::Platform.windows? ? 'nfs' : nil
-  config.vm.synced_folder '.', '/vagrant', type: synced_folder_type
+  config.vm.synced_folder '.', '/vagrant', type: :nfs
```
Comment (cody-vibe) [inline Vagrantfile]: This is not a good choice, nfs was a windows specific workaround, and it makes life harder on linux, requiring nfsd to be started and a root password any time the vm is woken up.

File: docs/host_setup.md [hunk 2]
```diff
@@ -6,100 +6,14 @@
...
-cd redmine_docker/project
-
-# Using HTTPS (not recommended, please use the SSH method below):
-git clone https://github.com/redmine/redmine.git
-
-# Using SSH
-git clone git@github.com:redmine/redmine.git
-```
-
-
```
Comment (cody-vibe) [inline docs/host_setup.md]: I'm going to keep these as a reference for the provision playbook.

File: project/config/Gemfile.local [hunk 3]
```diff
@@ -10,4 +10,4 @@ group :development do
 end
 
 # The awesome web server
-gem 'puma', '3.6.2'
+# gem 'puma', '3.6.2'
```
Comment (cody-vibe) [inline project/config/Gemfile.local:13]: I'l look into a way to have puma here and not interfere with redmine master.

File: project/docker/services/web/Dockerfile.production [hunk 4]
```diff
@@ -24,6 +24,7 @@ RUN set -ex                                                 && \
       git                                                      \
       libmagickwand-dev                                        \
       libpq-dev                                                \
+      subversion                                               \
```
Comment (cody-vibe) [inline project/docker/services/web/Dockerfile.production:27]: Is this really necessary? Any plugin or dependency that needs to be cloned via svn?

Comment (marius-balteanu) [reply project/docker/services/web/Dockerfile.production:27]: Yes, it is useful when you want to test the repository module from Redmine.

**Comment by cody-vibe** [comment]
Closing this since I don't intend to extend it anymore.

**Pull request closed without merging**